kernel/kernel-api/src/main/java/io/delta/kernel/internal/hook/CheckpointHook.java

```diff
@@ -33,12 +34,20 @@ public CheckpointHook(Path tablePath, long checkpointVersion) {
   }
 
   @Override
-  public void threadSafeInvoke(Engine engine) throws IOException {
+  public void executeHook(Engine engine) throws IOException {
```

Comment:
This is now a breaking change because we're changing the name of this method. I think either:

a) We think engines can still invoke hooks "directly" (i.e. not via `threadSafeInvoke` on `TimedPostCommitHook`), in which case we should keep this method still called `threadSafeInvoke` and public, or

b) We _always_ want to go by `TrimedPostCommitHook`, in which case let's not have this method be public anymore.